code/game/gamemodes/miniantags/demons/slaughter_demon/slaughter_demon.dm

```diff
@@ -0,0 +1,325 @@
+	var/playstyle_string = "<B>You are the Slaughter Demon, a terrible creature from another existence. You have a single desire: to kill. \
+						You may use the blood crawl icon when on blood pools to travel through them, appearing and disappearing from the station with a short delay. \
+						While blood crawling you will heal back to half your health pool. Slowly. \
+						If sink into your blood while dead or critical mobs are around you, you will pull them in with you, allowing you to feast. Healing 100HP each. \
+						A bonus of 10 maximum hp is given per sentient humanoid consumed, every fourth corpse eaten will increase health by another 50. \
+						Attacking sentient humanoids grants a 5% damage increase and a 5% damage reduction, stacking up to 50% extra. Maxing out at 45 damage. You will loose this buff out of combat however.<B>"
+	deathmessage = "screams in anger as it collapses into a puddle of viscera!"
+
+/mob/living/simple_animal/demon/slaughter_demon/New()
+	..()
+	remove_from_all_data_huds()
+	var/datum/spell/bloodcrawldemon/bloodspell = new
+	var/datum/spell/demon_charge/dc = new
+	var/datum/spell/demon_slam/ds = new
+	charge_holder = dc
+	slam_holder = ds
+	AddSpell(dc)
+	AddSpell(ds)
+	AddSpell(bloodspell)
+	if(istype(loc, /obj/effect/dummy/slaughter_demon))
+		phased = FALSE
+	addtimer(CALLBACK(src, PROC_REF(attempt_objectives)), 5 SECONDS)
+
+/mob/living/simple_animal/demon/slaughter_demon/proc/attempt_objectives()
+	if(!mind)
+		return
+	else
+		var/list/messages = list()
+		messages.Add(playstyle_string)
+		messages.Add("<b><span class ='notice'>You are not currently in the same plane of existence as the station. Use the blood crawl action at a blood pool to manifest.</span></b>")
+		SEND_SOUND(src, sound('sound/misc/demon_dies.ogg'))
+		if(!vialspawned)
+			SSticker.mode.traitors |= mind
+			mind.add_mind_objective(/datum/objective/slaughter)
+			mind.add_mind_objective(/datum/objective/demon_fluff)
+			messages.Add(mind.prepare_announce_objectives(FALSE))
+		messages.Add("<span class='motd'>For more information, check the wiki page: ([GLOB.configuration.url.wiki_url]/index.php/Slaughter_Demon)</span>")
+		to_chat(src, chat_box_red(messages.Join("<br>")))
+
+/obj/effect/dummy/slaughter_demon //Can't use the wizard one, blocked by jaunt/slow
+	name = "odd blood"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	density = FALSE
+	anchored = TRUE
+	invisibility = 60
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+
+/obj/effect/dummy/slaughter_demon/relaymove(mob/user, direction)
+	forceMove(get_step(src, direction))
+
+/obj/effect/dummy/slaughter_demon/ex_act()
+	return
+
+/obj/effect/dummy/slaughter_demon/bullet_act()
+	return
+
+/obj/effect/dummy/slaughter_demon/singularity_act()
+	return
+
+/mob/living/simple_animal/demon/slaughter_demon/Destroy()
+	for(var/mob/living/M in consumed_mobs)
+		REMOVE_TRAIT(M, TRAIT_UNREVIVABLE, "demon")
```

Comment:
why not move this into `relesae_consumed()` so this trait is auto removed when the mob is released?